actor/builtin/miner/miner.go

```diff
@@ -461,6 +465,36 @@ func (ma *Actor) IsBootstrapMiner(ctx exec.VMContext) (bool, uint8, error) {
 	return ma.Bootstrap, 0, nil
 }
 
+const (
+	WithinProvingPeriod = iota
+	AfterProvingPeriod
+	AfterGenerationAttackThreshold
+)
+
+// GetPoStState returns whether the miner's last submitPoSt within the proving period,
+// late or after the generation attack threshold.
```

Comment:
This method should only check whether the PoSt is late if the miner is actively proving storage (the ProvingSet is non-empty). Otherwise this method will suggest the miner is late when they are not.